lib/API/JSON/API.php

```diff
@@ -26,10 +26,17 @@ class API {
 
   /** @var AccessControl */
   private $access_control;
+  
+  /** @var WPFunctions */
+  private $wp;
 
   const CURRENT_VERSION = 'v1';
 
-  function __construct(ContainerInterface $container, AccessControl $access_control) {
+  function __construct(ContainerInterface $container, AccessControl $access_control, WPFunctions $wp = null) {
+    if($wp === null) {
```

Comment:
Why are you doing this? Why don't you let DI container do its job? It is done in many places in this Pull Request. I will not comment on every single one.  It is not possible everywhere, but it should not be nullabe here.

Reply:
You are right, I fixed this case, but other cases are not handled by DI yet.